test/expected/hnsw_vector.out

```diff
@@ -242,3 +242,108 @@ SELECT id FROM small_world_arr ORDER BY v <-> ARRAY[0,0,0];
   3
 (3 rows)
 
+-- Test pgvector in lantern.pgvector_compat=TRUE mode
+DROP TABLE small_world;
+\ir utils/small_world_vector.sql
+CREATE TABLE small_world (
+    id VARCHAR(3),
+    b BOOLEAN,
+    v VECTOR(3)
+);
+INSERT INTO small_world (id, b, v) VALUES
+    ('000', TRUE,  '[0,0,0]'),
+    ('001', TRUE,  '[0,0,1]'),
+    ('010', FALSE, '[0,1,0]'),
+    ('011', TRUE,  '[0,1,1]'),
+    ('100', FALSE, '[1,0,0]'),
+    ('101', FALSE, '[1,0,1]'),
+    ('110', FALSE, '[1,1,0]'),
+    ('111', TRUE,  '[1,1,1]');
+-- Distance functions
+SET lantern.pgvector_compat=TRUE;
+SET enable_seqscan=OFF;
+-- Note:
+-- For l2sqs and cosine distances in SELECT statement 
+-- It is better to use the function by name like cos_dist or l2sq_dist
+-- As operators for vector types are provided from pgvector, so it may cause undefined behaviour
```

Comment:
they are provided from pgvector for the vector type, right?
so, if they are used on real[] the real[] would first be converted to vector, implicitly, then the operator would be applied.

I think this should always be well defined (maybe not very perfromant). Am I missing something?